### PR TITLE
CONTRIB-8053: List Table Pagination Bugfix  and CONTRIB-8052: Course List Improvement

### PR DESCRIPTION
**Pull request by caglaroflazoglu**
Pagination does not work properly on 25x

Example: List count is 25:

`$pages = floor(count($meetingidsarray) / 25) + 1;`

$pages --> floor(1)+1 => 2  (wrong, $pages number must be 1)

### CODE REVIEW

File: locallib.php [hunk 1]
```diff
@@ -2182,7 +2187,7 @@ function bigbluebuttonbn_import_get_courses_for_select(array $bbbsession) {
     }
     $coursesforselect = [];
     foreach ($courses as $course) {
-        $coursesforselect[$course->id] = $course->fullname;
+        $coursesforselect[$course->id] = $course->fullname . " (" . $course->shortname . ")";
```
Comment (caglaroflazoglu) [inline locallib.php:2190]: Second fix is about same course name problem. I added short name after course name

File: locallib.php [hunk 2]
```diff
@@ -248,9 +248,14 @@ function bigbluebuttonbn_get_recordings_array_fetch($meetingidsarray) {
...
     // Execute a paginated getRecordings request.
-    $pages = floor(count($meetingidsarray) / 25) + 1;
+    $pagecount = 25;
+    $pages = floor(count($meetingidsarray) / $pagecount) + 1;
+
+    if(count($meetingidsarray) > 0 &&  count($meetingidsarray) % $pagecount == 0)
+        $pages--;
+
     for ($page = 1; $page <= $pages; ++$page) {
-        $mids = array_slice($meetingidsarray, ($page - 1) * 25, 25);
```
Comment (caglaroflazoglu) [inline locallib.php:253]: // $mids will empty array when $page is 2;

```
function bigbluebuttonbn_get_recordings_array_fetch_page($mids) 
...
    $url = \mod_bigbluebuttonbn\locallib\bigbluebutton::action_url('getRecordings', ['meetingID' => implode(',', $mids)]); // meetingID=??? is empty and unstable usage
...
```

Comment (caglaroflazoglu) [reply locallib.php:253]: travis-ci error fixed: Inline control structures are not allowed (line 253-255)

**Comment by jfederico** [comment]
Thanks for the PR

**Pull request merged**